src/test/java/com/automationpractice/MainPageFeatureSectionTests.java

```diff
@@ -104,57 +122,152 @@ public void shouldDisplayMoreForFirstItemOnPopularSectionWhenMouseOver() {
 
     @Test
     public void shouldDisplayPriceForFirstItemOnPopularSectionWhenMouseOver() {
-
+        mainPage.getPopularLink().click();
+        mainPage.hoverMouseOverFirstPopularElement();
+        Assertions.assertTrue(mainPage.getPriceFromFirstItemWhenMouseOver().isDisplayed());
+        String prize = "$16.51";
+        Assertions.assertEquals(prize, mainPage.getPriceFromFirstItemWhenMouseOver().getText());
     }
 
     @Test
-    public void shouldDisplayPopupAfterClickingOnFirstItemOnPopularSection() {
-
+    public void shouldDisplayTitleForSecondItemOnPopularSection() {
+        mainPage.getPopularLink().click();
+        String defaultLink = "http://automationpractice.com/img/p/7/7-home_default.jpg";
+        Assertions.assertEquals(defaultLink, mainPage.getPopularOffers().get(1).findElement(By.className("replace-2x")).getAttribute("src"));
     }
 
     @Test
-    public void shouldDisplayMainImageOnViewSection() {
-
+    public void shouldDisplayPriceForSecondItemOnPopularSection() {
+        mainPage.getPopularLink().click();
+        String price = mainPage.getPriceForItemsOnPopularSection().get(1).findElement(By.className("price")).getText();
+        Assertions.assertEquals("$27.00", price);
     }
 
     @Test
-    public void shouldDisplayBoxOfImagesOnViewSection() {
+    public void shouldDisplayQuickViewOnSecondPopularItemWhenMouseOver() {
+        mainPage.getPopularLink().click();
+        mainPage.hoverMouseOverSecondPopularElement();
+        Assertions.assertEquals(1, mainPage.getQuickViewLinks().size());
+        Assertions.assertTrue(mainPage.getQuickViewLinkOnSecondPopularItem().isDisplayed());
+        Assertions.assertEquals("Quick view", mainPage.getQuickViewLinkOnSecondPopularItem().getText());
+    }
 
+    @Test
+    public void shouldDisplayAddToCartForSecondItemOnPopularSectionWhenMouseOver() {
+        mainPage.getPopularLink().click();
+        mainPage.hoverMouseOverSecondPopularElement();
+        Assertions.assertTrue(mainPage.getButtonAddToCartOnSecondPopularItem().isDisplayed());
+        String buttonAddToCart = "Add to cart";
+        Assertions.assertEquals(buttonAddToCart, mainPage.getButtonAddToCartOnSecondPopularItem().getText());
     }
 
     @Test
-    public void shouldDisplayOtherViewAfterClickingArrowLeftOnViewSection() {
+    public void shouldDisplayMoreForSecondItemOnPopularSectionWhenMouseOver() {
+        mainPage.getPopularLink().click();
+        mainPage.hoverMouseOverSecondPopularElement();
+        Assertions.assertTrue(mainPage.getButtonMoreForSecondItemOnPopularSection().isDisplayed(), "Button 'More' is not displayed");
+        String buttonMore = "More";
```

Comment:
This could be inlined...

Reply:
Yes, that would be more clear. I changed that in all places.